### PR TITLE
Sync binary-search docs with problem-specifications

### PR DESCRIPTION
**Pull request by kytrinyx**
The binary-search exercise has been overhauled as part of a project to make practice exercises more consistent and friendly.

For more context, please see the discussion in the forum, as well as the pull request that updated the exercise in the problem-specifications repository:

- https://forum.exercism.org/t/new-project-making-practice-exercises-more-consistent-and-human-across-exercism/3943
- https://github.com/exercism/problem-specifications/pull/2220

### CODE REVIEW

File: exercises/practice/binary-search/.docs/instructions.md
Comment (kotp) [inline exercises/practice/binary-search/.docs/instructions.md:14]: We could emphasize the "sorted" as it is a requirement of the algorithm.

```suggestion
- Divide the _sorted_ list in half and compare the middle element with the item we're looking for.
```

File: exercises/practice/binary-search/.docs/introduction.md
Comment (kotp) [inline exercises/practice/binary-search/.docs/introduction.md:4]: Should we be more explicit in which meaning of "number" we mean?   When I first read this, I took it as the musical number, rather than a numeric number, even having seen that they were mathematicians doing this.

Then I had to think about what it meant, was it referring to the ditties that they liked, or their favorite scores, or how did they separate those arrangements.  Then, and only after, did I realize that in this context, we must mean numeric numbers, not musical numbers.

For some reason the music subject took over the context of the vocabulary used here.  Even after having read about numeric numbers in this review in the instructions file.

**Review by kotp — APPROVED** [review]
Approving, with a suggestion of emphasizing the sorted requirement.

And mentioning the different meanings of the word `number` and that I might be strange (hmpfh) but it took me some effort to realize what we were talking about.

File: exercises/practice/binary-search/.docs/introduction.md
Comment (kytrinyx) [reply exercises/practice/binary-search/.docs/introduction.md:4]: We're going to need to make this change upstream, rather than here. I will open a PR tomorrow!

File: exercises/practice/binary-search/.docs/instructions.md
Comment (kytrinyx) [reply exercises/practice/binary-search/.docs/instructions.md:14]: We have an `exercism/caution` block right above. I _think_ (but am not 100% confident) that this should be enough to not need emphasizing within the sentence itself.

What do you think?

Comment (kotp) [reply exercises/practice/binary-search/.docs/instructions.md:14]: It could very well be enough if it is there.  It is also repeated, and so the repetition itself is an emphasis.  A little bit of markdown for emphasis could help to bring that up as being an important requirement though.  If we were speaking, I would inflect on that single word, just a little bit.

File: exercises/practice/binary-search/.docs/introduction.md
Comment (iHiD) [reply exercises/practice/binary-search/.docs/introduction.md:4]: Yeah, for both this and the other comment about emphasing the word, they'd need to happen upstream not here. I don't think the emphasising one would be worth repropogating across all 65 tracks as that's lots of noise for maintainers, but I think this one might be.

Comment (kotp) [reply exercises/practice/binary-search/.docs/introduction.md:4]: If we take care of both at the same time, since the change is coming, would it be an appropriate time for it?  Perhaps in two commits, so that if a track wanted one but not the other, they could choose to do so.

Comment (kytrinyx) [reply exercises/practice/binary-search/.docs/introduction.md:4]: The changes from problem-specifications get added to the tracks by using configlet sync. They get all of the changes or none of the changes.

I won't be making hand-crafted PRs (or even lovingly scripted PRs) to allow people to cherry-pick an emphasis that does not meaningfully change the instructions.

I feel quite strongly that the only good reason to not accept a PR that syncs from problem-specs is if the instructions **contradict** the exercise in that track.

Comment (kytrinyx) [reply exercises/practice/binary-search/.docs/introduction.md:4]: @kotp I have opened the PR here: https://github.com/exercism/problem-specifications/pull/2226

If you like, feel free to suggest a PR to change the emphasis.

There is another PR open to the same exercise as well, so I am happy to wait to put in the new PRs to the tracks until all three PRs have been merged.

File: exercises/practice/binary-search/.docs/instructions.md
Comment (kytrinyx) [reply exercises/practice/binary-search/.docs/instructions.md:14]: This is what it looks like on the site, for context.

Screen Shot 2023-03-08 at 12 12 45 PM

File: exercises/practice/binary-search/.docs/introduction.md
Comment (kotp) [reply exercises/practice/binary-search/.docs/introduction.md:4]: Opened a [PR][pr] for the emphasis.  (Also, yes, the asterisks for bold rather than italics for emphasis, @iHiD .

[pr]:  exercism/problem-specifications#2227

**Pull request merged**
